pkg/types/relayer.go

```diff
@@ -100,10 +98,29 @@ type ChainService interface {
 	Replay(ctx context.Context, fromBlock string, args map[string]any) error
 }
 
+type EVMService interface {
+	CallContract(ctx context.Context, msg *evm.CallMsg, confidence primitives.ConfidenceLevel) ([]byte, error)
+	GetLogs(ctx context.Context, filterQuery evm.FilterQuery) ([]*evm.Log, error)
+	BalanceAt(ctx context.Context, account string, blockNumber *big.Int) (*big.Int, error)
+	EstimateGas(ctx context.Context, call *evm.CallMsg) (uint64, error)
+	TransactionByHash(ctx context.Context, hash string) (*evm.Transaction, error)
+	TransactionReceipt(ctx context.Context, txHash string) (*evm.Receipt, error)
+
+	// GetTransactionFee retrieves the fee of a transaction in wei from the underlying chain's TXM
+	GetTransactionFee(ctx context.Context, transactionID string) (*TransactionFee, error)
+	LatestAndFinalizedHead(ctx context.Context) (latest Head, finalized Head, err error)
```

Comment:
We already have `ChainService.LatestHead`. Should this one be on the generic `ChainService` too? Or alternatively, will this one diverge, and so should it be returning a custom evm head type instead of the shared one?

Reply:
This API must be stable by changing only in compatible ways, so a decision needs to be made now in this PR. If the intention is for this to be an evm specific method that returns an evm specific head, then it must start that way now.